serde/src/java/org/apache/hadoop/hive/serde2/io/HiveBaseCharWritable.java

```diff
@@ -26,10 +26,19 @@
 import org.apache.hive.common.util.HiveStringUtils;
 
 public abstract class HiveBaseCharWritable {
-  protected Text value = new Text();
+  protected Text value;
   protected int charLength = -1;
 
   public HiveBaseCharWritable() {
+    value = new Text();
+  }
+
+  public HiveBaseCharWritable(Text text) {
+    value = text;
+  }
+
+  public HiveBaseCharWritable(byte[] bytes) {
+    value = new Text(bytes);
```

Comment:
We could add an additional `set` method in sublcasses and no touch at all this class (fewer changes in total). Is there a specific reason for using the constructor approach?

Reply:
Likely no specific reason, it just happened to end up that way. I suppose that I didn't notice the set method on the Text object and wanted to avoid an extra allocation.